macantine/extract_open_data.py

```diff
@@ -171,6 +172,41 @@ def datetimes_to_str(df):
     return df
 
 
+def update_datagouv_resources():
+    """
+    Updating the URL of the different resources dsiplayed on data.gouv.fr in order to force their cache reload and display the correct update dates
+    Returns : Number of updated resources
+    """
+    if os.environ.get("ENVIRONMENT") != "prod":
+        return
+    dataset_id = os.getenv("DATAGOUV_DATASET_ID", "")
+    api_key = os.getenv("DATAGOUV_API_KEY", "")
+    if not (dataset_id and api_key):
+        logger.error("Datagouv resource update : API key or dataset id incorrect values")
+        return
+    header = {"X-API-KEY": api_key}
+    try:
+        response = requests.get(f"https://www.data.gouv.fr/api/1/datasets/{dataset_id}", headers=header)
+        response.raise_for_status()
+        resources = response.json()["resources"]
+        count_updated_resources = 0
+        for resource in resources:
+            if resource["format"] in ["xlsx", "csv"]:
+                today = date.today()
+                updated_url = resource["url"].split("?v=")[0] + "?v=" + today.strftime("%Y%m%d")
+                response = requests.put(
+                    f'https://www.data.gouv.fr/api/1/datasets/{dataset_id}/resources/{resource["id"]}',
+                    headers=header,
+                    json={"url": updated_url},
+                )
+                response.raise_for_status()
```

Comment:
Si je comprends bien, l'échec d'un des ressources lance une exception dans cette ligne, qui est catché par l'except en L:205 mettant fin à la boucle et à la méthode.

Je suis OK avec ça, je veux juste m'assurer que c'est bien le comportement attendu et qu'on ne voudrait pas continuer avec les autres ressources de la boucle for (dans ce cas il faudrait un try-catch à l'intérieur de la boucle for, mais c'est peut-être overkill)